common/short-hex-str/src/lib.rs

```diff
@@ -89,6 +90,7 @@ fn byte2hex(byte: u8) -> (u8, u8) {
 /// Hex encode a byte slice into the destination byte slice.
 #[inline(always)]
 fn hex_encode(src: &[u8], dst: &mut [u8]) {
+    #[allow(clippy::integer_arithmetic)]
     debug_checked_precondition!(dst.len() == 2 * src.len());
```

Comment:
this is a MIRAI annotation that will only happen during debug mode, so you can also do this:

```
debug_checked_precondition!(dst.len() == src.len().checked_mul(2).expect("should not happen"));
```

and this will not panic in release mode